hundred_x/async_client.py

```diff
@@ -1,29 +1,230 @@
+            **self.get_shared_params(),
+        )
+        response = await self.send_message_to_endpoint("/v1/order", "POST", message)
+        return response
 
-    async def get_server_time(self):
+    async def cancel_order(self, product_id: int, order_id: int):
+        """
+        Cancel an order.
+        """
+        message = self.generate_and_sign_message(
+            CancelOrder,
+            subAccountId=self.subaccount_id,
+            productId=product_id,
+            orderId=order_id,
+            **self.get_shared_params(),
+        )
+        response = await self.send_message_to_endpoint("/v1/order", "DELETE", message)
+        return response
+
+    async def cancel_and_replace_order(
+        self,
+        subaccount_id: int,
+        product_id: int,
+        quantity: int,
+        price: int,
+        side: OrderSide,
+        order_id_to_cancel: str,
+        nonce: int = 0,
+    ):
+        """
+        Cancel and replace an order.
+        """
+        ts = self._current_timestamp()
+        if nonce == 0:
+            nonce = ts
+        _message = self.generate_and_sign_message(
+            Order,
+            subAccountId=subaccount_id,
+            productId=product_id,
+            quantity=int(Decimal(str(quantity)) * Decimal(1e18)),
+            price=int(Decimal(str(price)) * Decimal(1e18)),
+            isBuy=side.value,
+            orderType=OrderType.LIMIT_MAKER.value,
+            timeInForce=TimeInForce.GTC.value,
+            nonce=nonce,
+            expiration=(ts + 1000 * 60 * 60 * 24) * 1000,
+            **self.get_shared_params(),
+        )
+        message = {}
+        message["newOrder"] = from_message_to_payload(_message)
+        message["idToCancel"] = order_id_to_cancel
+        response = await self.send_message_to_endpoint("/v1/order/cancel-and-replace", "POST", message)
+        return response
+    
+    async def cancel_all_orders(self, subaccount_id: int, product_id: int):
         """
-        Get the server time.
+        Cancel all orders.
         """
-        return super().get_server_time()
+        message = self.generate_and_sign_message(
+            CancelOrders,
+            subAccountId=subaccount_id,
+            productId=product_id,
+            **self.get_shared_params(),
+        )
+        response = await self.send_message_to_endpoint("/v1/openOrders", "DELETE", message)
+        return response
+    
+    async def login(self):
+        """
+        Login to the exchange.
+        """
+        response = await self.create_authenticated_session_with_service()
+        if response is None:
+            raise Exception("Failed to login")
+
+    async def send_message_to_endpoint(self, endpoint: str, method: str, message: dict = {}, authenticated: bool = True):
+        """
+        Send a message to an endpoint.
+        """
+        payload = from_message_to_payload(message)
+        async with httpx.AsyncClient() as client:
```

Comment:
im not sure why you would introduce an extra dependency for this.

Other than that its really good and i would merge it.

Would you like me to remove the httpx  dependency or would you prefer to do it?